### PR TITLE
inkscape: 1.1.2 -> 1.2

### PR DESCRIPTION
**Pull request by matthuszagh**
https://inkscape.org/news/2022/05/16/inkscape-12/
https://media.inkscape.org/media/doc/release_notes/1.2/Inkscape_1.2.html

```
$ nix-review rev HEAD                               
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
remote: Enumerating objects: 65, done.
remote: Counting objects: 100% (63/63), done.
remote: Compressing objects: 100% (25/25), done.
remote: Total 65 (delta 34), reused 53 (delta 30), pack-reused 2
Unpacking objects: 100% (65/65), 176.16 KiB | 2.17 MiB/s, done.
From https://github.com/NixOS/nixpkgs
   34e4df55664c..ea69f7982997  master     -> refs/nixpkgs-review/0
$ git worktree add /home/matt/.cache/nixpkgs-review/rev-2af19ca97cfa9eb904cf4db893163d55b95cf357/nixpkgs ea69f79829972187ecf6c1eaede4ce7a15d81736
Preparing worktree (detached HEAD ea69f7982997)
Updating files: 100% (30489/30489), done.
HEAD is now at ea69f7982997 Merge pull request #172988 from McSinyx/dendrite-0.8.5
$ nix-env --option system x86_64-linux -f /home/matt/.cache/nixpkgs-review/rev-2af19ca97cfa9eb904cf4db893163d55b95cf357/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff 2af19ca97cfa9eb904cf4db893163d55b95cf357
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-linux -f /home/matt/.cache/nixpkgs-review/rev-2af19ca97cfa9eb904cf4db893163d55b95cf357/nixpkgs -qaP --xml --out-path --show-trace --meta
38 packages updated:
adapta-gtk-theme arc-theme arx-libertatis asciidoc-full asciidoc-full-with-plugins ayu-theme-gtk-unstable capitaine-cursors cinnamon-screensaver clevis dblatex-full disorderfs emojione fim gnome-documents inkscape (1.1.2 → 1.2) inkscape-applytransforms inkscape-with-extensions (1.1.2 → 1.2) iso-flags-unstable iso-flags-unstable iso-flags-unstable k40-whisperer kabeljau luksmeta mate-utils mojave-gtk-theme-unstable numix-cursor-theme numix-solarized-gtk-theme plata-theme pop-gtk-theme python3.10-diagrams python310Packages.inkex (1.1.2 → 1.2) python3.9-diagrams python39Packages.inkex (1.1.2 → 1.2) rep spring springlobby tang wpa_gui

$ nix --experimental-features nix-command build --no-link --keep-going --option build-use-sandbox relaxed -f /home/matt/.cache/nixpkgs-review/rev-2af19ca97cfa9eb904cf4db893163d55b95cf357/build.nix
error: builder for '/nix/store/pf7nk0iwaf61zmv57x1qxldrgbvarsgk-inkscape-applytransforms-0.pre+unstable=2021-05-11.drv' failed with exit code 2;
       last 10 log lines:
       > =============================== warnings summary ===============================
       > ../../nix/store/ry1qvmclhsi47c5h8g5agr32jwybjl06-python3.9-inkex-1.2/lib/python3.9/site-packages/inkex/elements/_svg.py:172
       >   /nix/store/ry1qvmclhsi47c5h8g5agr32jwybjl06-python3.9-inkex-1.2/lib/python3.9/site-packages/inkex/elements/_svg.py:172: DeprecationWarning: invalid escape sequence \s
       >     float(unit) for unit in re.split(",\s*|\s+", self.get("viewBox", "0"))
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > ERROR tests/test_applytransform.py
       > !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
       > ========================= 1 warning, 1 error in 0.15s ==========================
       For full logs, run 'nix log /nix/store/pf7nk0iwaf61zmv57x1qxldrgbvarsgk-inkscape-applytransforms-0.pre+unstable=2021-05-11.drv'.
error: builder for '/nix/store/rvw9lmy156yjzrxvm0wbhvf92q5l71gx-spring-105.0.1-1486-g8581792.drv' failed with exit code 2;
       last 10 log lines:
       >       |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       >    40 |                                   __va_arg_pack ());
       >       |                                   ~~~~~~~~~~~~~~~~~
       > [ 50%] Linking CXX static library libassimp.a
       > [ 50%] Built target assimp
       > [ 50%] Linking CXX shared library ../../libunitsync.so
       > [ 50%] Built target unitsync
       > [ 50%] Linking CXX executable ../../../spring-dedicated
       > [ 50%] Built target engine-dedicated
       > make: *** [Makefile:146: all] Error 2
       For full logs, run 'nix log /nix/store/rvw9lmy156yjzrxvm0wbhvf92q5l71gx-spring-105.0.1-1486-g8581792.drv'.
error: 1 dependencies of derivation '/nix/store/wd6vdxq6ksi9nkpw4hr3a2m5xivqndba-springlobby-0.270.drv' failed to build
error: 3 dependencies of derivation '/nix/store/nbg5cp7nama5n27mgab1wkpwvmi2d97l-review-shell.drv' failed to build
1 package marked as broken and skipped:
gnome.gnome-documents

3 packages failed to build:
inkscape-extensions.applytransforms spring springLobby

34 packages built:
adapta-gtk-theme arc-theme arx-libertatis asciidoc-full asciidoc-full-with-plugins ayu-theme-gtk capitaine-cursors cinnamon.cinnamon-screensaver cinnamon.iso-flags-png-320x420 cinnamon.iso-flags-svg clevis dblatexFull disorderfs emojione fim inkscape inkscape-with-extensions iso-flags k40-whisperer kabeljau kakounePlugins.rep luksmeta mate.mate-utils mojave-gtk-theme numix-cursor-theme numix-solarized-gtk-theme plata-theme pop-gtk-theme python310Packages.diagrams python310Packages.inkex python39Packages.diagrams python39Packages.inkex tang wpa_supplicant_gui
```

These errors don't seem to have anything to do with inkscape; appear to be errors in the packages themselves. Unfortunately, inkex displays a different error after fixing the dependency.

I also test `result/bin/inkscape` and seemed to work correctly.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/inkscape/default.nix [hunk 1]
```diff
@@ -53,11 +53,14 @@ let
...
-  version = "1.1.2";
+  version = "1.2";
 
-  src = fetchurl {
-    url = "https://media.inkscape.org/dl/resources/file/${pname}-${version}.tar.xz";
-    sha256 = "sha256-P/5UoG0LJaTNi260JFNu8e0gW+E0Q6Oc1DfIx7ibltE=";
+  src = fetchFromGitLab {
+    owner = "inkscape";
+    repo = "inkscape";
+    rev = "dc2aedaf03f19bd890529a9e4720a02465fa1b6c";
```
Comment (jtojnar) [inline pkgs/applications/graphics/inkscape/default.nix:61]: You would need to use `INKSCAPE_${builtins.replaceStrings [ "."] [ "_"] version}`. But it is better to wait for the tarball release as `fetchSubmodules` can sometimes be quite brittle.

**Comment by jtojnar** [comment]
The clipart importer has been re-added, though now it seems to use `requests` so we should not need to re-add glib-networking:

https://gitlab.com/inkscape/extras/inkscape-import-clipart/-/blob/ad94a095f76653b0dcfaff3eca4a6e72b8a6028e/import_sources.py#L229

Should be tested either way.

File: pkgs/applications/graphics/inkscape/default.nix (hunk 1 above)
Comment (matthuszagh) [reply pkgs/applications/graphics/inkscape/default.nix:61]: I attempted fetching the tar.gz source tarball. It seems the submodules aren't included it.

Comment (jtojnar) [reply pkgs/applications/graphics/inkscape/default.nix:61]: Do you mean the official tarball, or the GitLab one? The latter is useless when project contains submodules, just like on GitHub. If the former, then that needs to be reported upstream.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 173184` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.diagrams</li>
    <li>python39Packages.diagrams</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>asciidoc-full-with-plugins</li>
    <li>emojione</li>
    <li>inkscape-extensions.applytransforms</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>asciidoc-full</li>
    <li>dblatexFull</li>
    <li>inkscape</li>
    <li>inkscape-with-extensions</li>
    <li>k40-whisperer</li>
    <li>kabeljau</li>
    <li>kakounePlugins.rep</li>
    <li>numix-cursor-theme</li>
    <li>python310Packages.inkex</li>
    <li>python39Packages.inkex</li>
  </ul>
</details>

File: pkgs/applications/graphics/inkscape/default.nix (hunk 1 above)
Comment (matthuszagh) [reply pkgs/applications/graphics/inkscape/default.nix:61]: The latter. I'll hold out for the official one. Thanks for the heads up.

Comment (andersk) [reply pkgs/applications/graphics/inkscape/default.nix:61]: The [official tarball](https://media.inkscape.org/dl/resources/file/inkscape-1.2_2022-05-15_dc2aedaf03.tar.xz) linked from the [release page](https://inkscape.org/release/inkscape-1.2/) comes with submodules included.

Comment (matthuszagh) [reply pkgs/applications/graphics/inkscape/default.nix]: Updated.

File: pkgs/applications/graphics/inkscape/default.nix [hunk 2]
```diff
@@ -55,12 +55,9 @@ stdenv.mkDerivation rec {
   pname = "inkscape";
   version = "1.2";
 
-  src = fetchFromGitLab {
-    owner = "inkscape";
-    repo = "inkscape";
-    rev = "dc2aedaf03f19bd890529a9e4720a02465fa1b6c";
-    sha256 = "sha256-h10E33THDM2Ci4uQ+7/KEmmnm6orAeV8seKa3IYPCGc=";
-    fetchSubmodules = true;
+  src = fetchzip {
```
Comment (andersk) [inline pkgs/applications/graphics/inkscape/default.nix:58]: `src = fetchurl {…}` is much more common than `src = fetchzip {…}`. The manual [explains](https://nixos.org/manual/nixpkgs/stable/#fetchurl) the difference but not which to prefer. It makes sense to me that storing the source compressed (`fetchurl`) generally makes more sense than uncompressed (`fetchzip`)?

File: pkgs/applications/graphics/inkscape/default.nix [hunk 3]
```diff
@@ -53,11 +53,11 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "inkscape";
-  version = "1.1.2";
+  version = "1.2";
 
   src = fetchurl {
-    url = "https://media.inkscape.org/dl/resources/file/${pname}-${version}.tar.xz";
-    sha256 = "sha256-P/5UoG0LJaTNi260JFNu8e0gW+E0Q6Oc1DfIx7ibltE=";
+    url = "https://media.inkscape.org/dl/resources/file/${pname}-${version}_2022-05-15_dc2aedaf03.tar.xz";
```
Comment (jtojnar) [inline pkgs/applications/graphics/inkscape/default.nix:59]:
```suggestion
    url = "https://media.inkscape.org/dl/resources/file/inkscape-${version}.tar.xz";
```

**Comment by DamienCassou** [comment]
Result of `nixpkgs-review pr 173184` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>gnome.gnome-documents</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>inkscape-extensions.applytransforms</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>33 packages built:</summary>
  <ul>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>ayu-theme-gtk</li>
    <li>capitaine-cursors</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>cinnamon.iso-flags-png-320x420</li>
    <li>cinnamon.iso-flags-svg</li>
    <li>clevis</li>
    <li>dblatexFull</li>
    <li>disorderfs</li>
    <li>emojione</li>
    <li>fim</li>
    <li>inkscape</li>
    <li>inkscape-with-extensions</li>
    <li>iso-flags</li>
    <li>k40-whisperer</li>
    <li>kabeljau</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>luksmeta</li>
    <li>mate.mate-utils</li>
    <li>mojave-gtk-theme</li>
    <li>numix-cursor-theme</li>
    <li>numix-solarized-gtk-theme</li>
    <li>plata-theme</li>
    <li>pop-gtk-theme</li>
    <li>python310Packages.diagrams</li>
    <li>python310Packages.inkex</li>
    <li>python39Packages.diagrams</li>
    <li>python39Packages.inkex</li>
    <li>tang</li>
    <li>wpa_supplicant_gui</li>
  </ul>
</details>

File: pkgs/applications/graphics/inkscape/default.nix [hunk 4]
```diff
@@ -73,20 +73,6 @@ stdenv.mkDerivation rec {
       # e.g., those from the "Effects" menu.
       python3 = "${python3Env}/bin/python";
     })
-
-    # Fix build with poppler 22.03
-    # https://gitlab.com/inkscape/inkscape/-/merge_requests/4187
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/inkscape/default.nix:79]: Please remove the unused fetchpatch from inputs.

**Review by jtojnar — APPROVED** [review]
I added few more missing dependencies and the clipart import now seems to work.

**Pull request merged**